### ISSUE TITLE
Add TextMate support to parse syntax highlighting without tree-sitter

### ISSUE THREAD

**Issue by HeavySnowJakarta** [zed-industries/zed#18320]
### Check for existing issues

- [X] Completed

### Describe the feature

It's not hard to understand tree-sitter is the first-class citizen, but there are cases that ASTs are not available. There is no tree-sitter parser for some languages (like cmd batch, powershell and specific assembly languages) and sometimes we don't want full language support (like temporarily viewing sources without installing its extension, and disabling tree-sitter parsing for lower memory occupation), and now the case is, without a tree-sitter parser, there is no syntax highlighting. For VSCode, common programming language syntax highlighting is built-in (easier and lighter), and with tree-sitter and language server we can have full support for that language.

**Comment by notpeter** [comment]
For reference, TextMate grammars use [kkos/oniguruma](https://github.com/kkos/oniguruma) flavored regular expressions so likely Zed would likely have to embed that, perhaps via the [onig](https://crates.io/crates/onig) crate.

**Comment by AndydeCleyre** [comment]
While this issue is for the TextMate syntax format in particular, I'll mention some alternatives which also have a wide array of existing languages defined:

- [pygments](https://pygments.org/languages/)
- [highlight](https://gitlab.com/saalen/highlight/-/tree/master/langDefs)

For me, this issue is one of the handful still making Sublime Text the superior editor.

Is this or something like it on the team's roadmap for sure, or might Zed remain tree-sitter-only, always?

**Comment by notpeter** [comment]
> Is this or something like it on the team's roadmap for sure, or might Zed remain tree-sitter-only, always?

No, this is not on our current roadmap, but that does not mean Zed will be tree-sitter-only forever.

Given that pygments requires python and highlight is written in C++, we would be unlikely to bundle those in core. It is conceivable that Zed could add an extension API to enable extensions to provide syntax highlighting (beyond tree-sitters) that could leverage local processes (e.g. Python, C++ platform binaries) as we do for LSPs, but to my knowledge the design for such an Extension API has not been explored.

Also wanted to mention [trishume/syntect](https://github.com/trishume/syntect), a Rust library for Sublime Text syntax definitions (uses onig under the hood).
